common/models/user.js

```diff
@@ -592,18 +609,30 @@ module.exports = function(User) {
   User.hashPassword = function(plain) {
     this.validatePassword(plain);
     var salt = bcrypt.genSaltSync(this.settings.saltWorkFactor || SALT_WORK_FACTOR);
-    return bcrypt.hashSync(plain, salt);
+    var hash = bcrypt.hashSync(plain, salt);
+    if (/^\$2a\$/.test(hash)) {
+      // change the prefix to indicate that we have verified the plain password
+      // is at most 72 chars long
+      hash = '$72$'.concat(hash.slice(4));
```

Comment:
Does the proposed fix reject passwords longer than 72 chars during login? I thought we only validate passwords for `User.create` and `User.resetPassword`.